hs_core/views/resource_metadata_rest_api.py

```diff
@@ -0,0 +1,217 @@
+import logging
+
+from rest_framework.response import Response
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework import status
+from rest_framework import generics
+from rest_framework import serializers
+
+from hs_core import hydroshare
+from hs_core.models import Contributor, CoreMetaData, Coverage, Creator, Date, \
+    ExternalProfileLink, Format, FundingAgency, Identifier, Subject, Source, Relation
+from hs_core.views import utils as view_utils
+from hs_core.views.utils import ACTION_TO_AUTHORIZE
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalProfileLinkSerializer(serializers.Serializer):
```

Comment:
I was thinking if you use ```ModelSerializer``` you won't have to define the model fields again within the serializer class. But if you have tried that and it is giving JSON error then it is fine the way you have implemented. I have never used ```ModelSerializer``` myself so don't know much about how it supposed to work.